server/src/utils/http_service.rs

```diff
@@ -281,31 +289,44 @@ async fn join_game(user_id: String, game_id: String, state: AsyncState) -> Vec<u
     .port();
 
   let mut inner_state = state.write().await;
-  inner_state
-    .games
-    .get_mut(&game_id)
-    .unwrap()
-    .players
-    .as_mut()
-    .unwrap()
-    .push(user_id.clone());
-
   inner_state
     .player_games
     .insert(user_id.clone(), game_id.clone());
 
-  let game = inner_state.games.get_mut(&game_id).unwrap();
+  let mut game = None;
   let mut builder = FlatBufferBuilder::new();
 
-  let game = game.pack(&mut builder);
+  let player = PlayerT  {
+    hp: 100,
+    id: Some(user_id.clone()),
+    look_direction: Some(Vec3T::default()),
+    position: Some(Vec3T::default()),
+    sneaking: false,
+    speed: 100,
+    sprinting: false,
+    role: Role::Default,
+    velocity: Some(Vec3T::default())
+  };
+
+  if let Some(game_state) = inner_state.game_states.get_mut(&game_id) {
+      let game_inner = game_state.game.as_ref();
+      game = Some(game_inner.expect("Game metadata not found").pack(&mut builder));
+
+      if let Some(players) = game_state.players.as_mut() {
+          players.push(player);
+      }
+  }
```

Comment:
The idea that the game object is a child of the game_state object seems a bit backwards to me. What are your thoughts on changing the FBS definitions such that the `Game` object has a `state: GameState` member instead?

Reply:
I did not think this fully through when I created the initial `.fbs` files.

Reply:
Hmm. I think that depends on the long term goals of the `Game` class. I know it's supposed to be metadata, and right now this includes things like the game phase and time. My programmatic take would be to question which class will need to access methods from the other. For instance, if the `Game` class is to be responsible for pausing the game, or switching phases etc, then it will need to perform actions on the `GameState` object, which would indeed mean inverting the relationship we have now.

On the other hand, the more I think about this, is there a world where we want to separate these two? I don't know that either object should know of the other's existence, so maybe we want to have a `Game` flatbuffer which houses a `GameMetadata` flatbuffer and a `GameState` flatbuffer. I think in terms of long term scalability, this make the most sense to me. Mainly because I do agree, having `Game` be a child of `GameState` seems quite weird. But having `GameState` be a child of `Game` also seems to violate the whole purpose of the `Game` object -- to be metadata.

Reply:
Agreed on your second point. I think it may be best to separate the two entirely and have the `game_state` object be entirely owned by the `web_rtc_service` similar to how the `lobby` object is essentially owned by the `chat_service`